packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -13,38 +13,86 @@ import 'pointer_router.dart';
 
 /// Signature for listening to [PointerEnterEvent] events.
 ///
-/// Used by [MouseTrackerAnnotation], [Listener] and [RenderPointerListener].
+/// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerEnterEventListener = void Function(PointerEnterEvent event);
 
 /// Signature for listening to [PointerExitEvent] events.
 ///
-/// Used by [MouseTrackerAnnotation], [Listener] and [RenderPointerListener].
+/// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerExitEventListener = void Function(PointerExitEvent event);
 
 /// Signature for listening to [PointerHoverEvent] events.
 ///
-/// Used by [MouseTrackerAnnotation], [Listener] and [RenderPointerListener].
+/// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerHoverEventListener = void Function(PointerHoverEvent event);
 
 /// The annotation object used to annotate layers that are interested in mouse
 /// movements.
 ///
-/// This is added to a layer and managed by the [Listener] widget.
+/// This is added to a layer and managed by the [MouseRegion] widget.
 class MouseTrackerAnnotation {
   /// Creates an annotation that can be used to find layers interested in mouse
   /// movements.
   const MouseTrackerAnnotation({this.onEnter, this.onHover, this.onExit});
 
-  /// Triggered when a pointer has entered the bounding box of the annotated
-  /// layer.
+  /// Triggered when a mouse pointer, with or without buttons pressed, has
+  /// entered the annotated region.
+  ///
+  /// This callback is triggered when the pointer has started to be contained
+  /// by the annotationed region for any reason.
+  ///
+  /// More specifically, the callback is triggered by the following cases:
+  ///
+  ///  * A new annotated region has appeared under a pointer.
+  ///  * An existing annotated region has moved to under a pointer.
+  ///  * A new pointer has been added to somewhere within an annotated region.
+  ///  * An existing pointer has moved into an annotated region.
+  ///
+  /// This callback is not always matched by an [onExit].
+  ///
+  /// See also:
+  ///
+  ///  * [MouseRegion.onEnter], which uses this callback.
+  ///  * [onExit], which is triggered when a mouse pointer exits the region.
   final PointerEnterEventListener onEnter;
 
-  /// Triggered when a pointer has moved within the bounding box of the
-  /// annotated layer.
+  /// Triggered when a pointer has moved within the annotated region without
+  /// buttons pressed.
+  ///
+  /// This is not triggered if the region under the pointer moves without
```

Comment:
Why the asymmetric?

Reply:
To clarify: (I'll add it to docs)
The `onHover` is triggered when
  - An annotation that did not contain the pointer moves under the pointer
  - A pointer hovers onto or within an annotation

The `onHover` is not triggered when
  - An annotation that is containing the pointer moves, and still contain the pointer

In the last case, the only thing that is changed is the `PointerHoverEvent.localPosition`, which is not correctly provided until https://github.com/flutter/flutter/issues/33675 is fixed. (I would like to finish that PR but it's actually a much bigger breaking change than imagined)

Reply:
Makes sense. I though we already had localPositions because in that case you do want to call onHover when an annotation moves under a pointer.

Can you please add a note to the bug to remind us that we need to change onHover when we support local positions?

Reply:
Ok, done. Thanks.